src/warm-cache-for-festival.ts

```diff
@@ -1,57 +1,134 @@
+import Ajv, {JSONSchemaType} from "ajv"
 import {existsSync, readFileSync, statSync} from "fs";
 import redis from "redis";
+import yaml from "yaml";
 
 import * as constants     from "./constants";
 import * as redisHelper   from "./redis-helper";
 import * as spotifyHelper from "./spotify-helper";
 
+interface Day {
+    day: string
```

Comment:
Since you're already parsing days into a known day variable (`for (const day of data.days)`) I think this should describe the actual field of the day object. If it's the former of what I mentioned in the previous comment, something like `dayIndex` or `index`, and if the latter `displayName`

Reply:
So I think there are probably a few representations here:

1. An index that can safely be used as the key in redis i.e. a 1-based index for festival:nosalive_2022:1
2. An actual ISO date (e.g. that could help to determine the start and end date of a festival by taking min/max across days)
3. A display name, for any festivals that assign names to dates e.g. Weekend 1 Saturday, Weekend 2 Sunday

So perhaps the fields become: index, date, displayName?

Reply:
That makes sense to me. I think date and displayName should be optional fields

Reply:
Commit updated. To avoid confusion, I named the "key" field _number_ rather than _index_ as I think index implies 0-based. So essentially, _number_ is used the same as the current text file format - 0 for no days, [1, 2, 3 ... n] otherwise. So it now looks like the below (date and display_name being optional):

```
...
days:
  - number: 1
    date: 19-Aug-22
    display_name: Weekend 1
    artists:
      - name: The Chemical Brothers
        spotify_uri: 1GhPHrq36VKCY3ucVaZCfo
      - name: The Strokes
  - number: 2
    date: 26-Aug-22
    display_name: Weekend 2
    artists:
      - name: The Cure
      - name: The XX
        spotify_uri: 3iOvXCl6edW5Um0fXEBRXZ
```

I'm not currently loading the date or display_name fields anywhere- want to take a look at that in a separate PR once have thought about a sensible proposal for storing in redis